src/app/core/data/data.service.ts

```diff
@@ -0,0 +1,747 @@
+import { HttpClient } from '@angular/common/http';
+import { Store } from '@ngrx/store';
+import { Operation } from 'fast-json-patch';
+import { AsyncSubject, combineLatest, from as observableFrom, Observable, of as observableOf } from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  find,
+  map,
+  mergeMap,
+  skipWhile,
+  switchMap,
+  take,
+  takeWhile,
+  tap,
+  toArray
+} from 'rxjs/operators';
+import { hasValue, isNotEmpty, isNotEmptyOperator } from '../../shared/empty.util';
+import { NotificationOptions } from '../../shared/notifications/models/notification-options.model';
+import { NotificationsService } from '../../shared/notifications/notifications.service';
+import { FollowLinkConfig } from '../../shared/utils/follow-link-config.model';
+import { getClassForType } from '../cache/builders/build-decorators';
+import { RemoteDataBuildService } from '../cache/builders/remote-data-build.service';
+import { RequestParam } from '../cache/models/request-param.model';
+import { ObjectCacheEntry } from '../cache/object-cache.reducer';
+import { ObjectCacheService } from '../cache/object-cache.service';
+import { DSpaceSerializer } from '../dspace-rest/dspace.serializer';
+import { HALEndpointService } from '../shared/hal-endpoint.service';
+import { getFirstCompletedRemoteData, getFirstSucceededRemoteData, getRemoteDataPayload } from '../shared/operators';
+import { URLCombiner } from '../url-combiner/url-combiner';
+import { ChangeAnalyzer } from './change-analyzer';
+import { PaginatedList } from './paginated-list.model';
+import { RemoteData } from './remote-data';
+import {
+  CreateRequest,
+  DeleteByIDRequest,
+  DeleteRequest,
+  GetRequest,
+  PatchRequest,
+  PostRequest,
+  PutRequest
+} from './request.models';
+import { RequestService } from './request.service';
+import { RestRequestMethod } from './rest-request-method';
+import { UpdateDataService } from './update-data.service';
+import { GenericConstructor } from '../shared/generic-constructor';
+import { NoContent } from '../shared/NoContent.model';
+import { CacheableObject } from '../cache/cacheable-object.model';
+import { CoreState } from '../core-state.model';
+import { FindListOptions } from './find-list-options.model';
+
+export abstract class DataService<T extends CacheableObject> implements UpdateDataService<T> {
```

Comment:
Please add a comment to describe this.  How does this differ from the BaseDataService we already have?  It looks like this is a copy of most of the methods from BaseDataService?  Is there a reason we don't just use BaseDataService

Reply:
I think DataService was just copied over and then forgot, is not used anywhere and all the methods are already present somewhere else. Deleted from code base.